pool/tick.gno

```diff
@@ -89,40 +70,65 @@ func (pool *Pool) tickUpdate(
 
 	if liquidityGrossBefore.IsZero() {
 		if tick <= tickCurrent {
-			info.feeGrowthOutside0X128 = feeGrowthGlobal0X128
-			info.feeGrowthOutside1X128 = feeGrowthGlobal1X128
+			thisTick.feeGrowthOutside0X128 = feeGrowthGlobal0X128
+			thisTick.feeGrowthOutside1X128 = feeGrowthGlobal1X128
 		}
 
-		info.initialized = true
+		thisTick.initialized = true
 	}
 
-	info.liquidityGross = liquidityGrossAfter
+	thisTick.liquidityGross = liquidityGrossAfter
 
 	if upper {
-		info.liquidityNet = i256.Zero().Sub(info.liquidityNet, liquidityDelta)
+		thisTick.liquidityNet = i256.Zero().Sub(thisTick.liquidityNet, liquidityDelta)
 	} else {
-		info.liquidityNet = i256.Zero().Add(info.liquidityNet, liquidityDelta)
+		thisTick.liquidityNet = i256.Zero().Add(thisTick.liquidityNet, liquidityDelta)
 	}
 
-	pool.ticks[tick] = info
+	pool.ticks[tick] = thisTick
 
 	return flipped
 }
 
 // tickCross updates a tick's state when it is crossed and returns the liquidity net.
 func (pool *Pool) tickCross(
 	tick int32,
-	feeGrowthGlobal0X128 *u256.Uint, // uint256
-	feeGrowthGlobal1X128 *u256.Uint, // uint256
+	feeGrowthGlobal0X128 *u256.Uint,
+	feeGrowthGlobal1X128 *u256.Uint,
 ) *i256.Int {
-	info := pool.ticks[tick]
-	info.init()
+	thisTick := pool.getTick(tick)
+
+	thisTick.feeGrowthOutside0X128 = new(u256.Uint).Sub(feeGrowthGlobal0X128, thisTick.feeGrowthOutside0X128)
+	thisTick.feeGrowthOutside1X128 = new(u256.Uint).Sub(feeGrowthGlobal1X128, thisTick.feeGrowthOutside1X128)
+
+	pool.ticks[tick] = thisTick
+
+	return thisTick.liquidityNet
+}
+
+func (pool *Pool) getTick(tick int32) TickInfo {
+	tickInfo := pool.ticks[tick]
+	tickInfo.init()
+
+	return tickInfo
+}
 
-	info.feeGrowthOutside0X128 = new(u256.Uint).Sub(feeGrowthGlobal0X128, info.feeGrowthOutside0X128)
-	info.feeGrowthOutside1X128 = new(u256.Uint).Sub(feeGrowthGlobal1X128, info.feeGrowthOutside1X128)
+func getFeeGrowthX128(
```

Comment:
I suggest this function should revert

This function is rather unreadable.
The traditional function seems more intuitive.
The conditional statement in this function tries to include the cases below and above, which makes it look more complicated with the separate conditional expressions.

Reply:
b69c6790916baf27d76fde2d936f99d0be5f7f20